File: Calculator.java
public class Calculator {
    public double calculateHandSpeed(RhythmType rhythm1, RhythmType rhythm2, double initalBpm){

        if (rhythm1 == null) {
            throw new IllegalArgumentException("Can't have null rhythm");
        }else if (rhythm2 == null) {
            throw new IllegalArgumentException("Can't have null rhythm");
        }        

        double newBPM;
        double desiredRhythm = rhythm1.getDuration();
        double actualRhythm = rhythm2.getDuration();

        if (initalBpm <= 0) {
            throw new IllegalArgumentException("Invalid BPM, choose a larger bpm");
        }else if (Double.isInfinite(initalBpm)) {
            throw new IllegalArgumentException("Invalid BPM, choose a smaller BPM");
        }

        newBPM = initalBpm * (actualRhythm / desiredRhythm);

        if (Double.isInfinite(newBPM)) {
            throw new IllegalArgumentException("New BPM too large");
        }else{
            return newBPM;
        }
    }
}

File: flutter/hand_speed_calculator/lib/logic/calculator.dart
import 'package:hand_speed_calculator/logic/rhythm_type.dart';

class Calculator {
  double calculateHandSpeed(RhythmType desiredRhythmParameter, RhythmType actualRhythmParameter, double initalBpm){
    final double newBPM;
    double desiredRhythm = desiredRhythmParameter.duration;
    double actualRhythm = actualRhythmParameter.duration;

    if (initalBpm <= 0) {
      throw ArgumentError("Invalid BPM, choose a larger bpm");
    }else if (initalBpm.isInfinite) {
      throw ArgumentError("Invalid BPM, choose a smaller BPM");
    }

    newBPM = initalBpm * (actualRhythm / desiredRhythm);

    if (newBPM.isInfinite) {
      throw ArgumentError("New BPM too large");
    }else if (newBPM <= 0) {
      throw ArgumentError("New bpm is less than 0");
    }else {
      return newBPM;
    }
  }
}
